[[ Bug 17962 ]] Ensure _mainStacksChanged isn't sent to a deleted stack

montegoulding opened the pull request:
When deleting a stack the default stack may be left
in a state where it references the deleted stack
which is an issue which needs further thought and
broader changes in a non-maintenance release. This
patch changes the target for `_mainStacksChanged`
which is always sent just after a stack is deleted
to the home stack. `_mainStacksChanged` is an IDE
only message so this change should have no impact
on users.

Commit:
[[ Bug 17962 ]] Ensure _mainStacksChanged isn't sent to a deleted stack

When deleting a stack the default stack may be left
in a state where it references the deleted stack
which is an issue which needs further thought and
broader changes in a non-maintenance release. This
patch changes the target for `_mainStacksChanged`
which is always sent just after a stack is deleted
to the home stack. `_mainStacksChanged` is an IDE
only message so this change should have no impact
on users.

peter-b commented:
@livecode-vulcan review ok f387e3a

There may be a small issue with the release notes but I'll fix that up when I review them before release.

livecode-vulcan commented:
💙 review by @peter-b ok f387e3a480556e4b88d91ec2f2217ac290a3a449

livecode-vulcan commented:
😎 test success f387e3a480556e4b88d91ec2f2217ac290a3a449
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2811)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2818)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2804)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2820)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2826)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2835)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2849)

montegoulding commented:
@peter-b if we want to include #4289 in rc-4 then this PR can be closed.

runrevmark commented:
@montegoulding: I suggest keeping this PR as well as it means that the message is always sent to a defined stack.

montegoulding commented:
@runrevmark ok

Commit:
[[ Bug 17962 ]] Fix default object resetting

This patch ensures that the defaultStack is valid
when resetting after a script may have deleted
the default stack.

Commit:
Merge branch 'bugfix-17962-b' into bugfix-17962

# Conflicts:
#	docs/notes/bugfix-17962.md

montegoulding commented:
Merged the two 17962 PRs because the note was the same filename

livecodepanos commented:
@livecode-vulcan review ok 9f60009

livecode-vulcan commented:
💙 review by @livecodepanos ok 9f600097f3146f6173e249b5b7861b57d77544d5

livecode-vulcan commented:
😎 test success 9f600097f3146f6173e249b5b7861b57d77544d5
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/5)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/5)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/5)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/5)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/5)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/5)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/5)
